picoapi/api.py

```diff
@@ -82,7 +83,7 @@ def __init__(
 
         else:
             # add the service registration event
-            self.router.on_event("startup", register_uservice)
+            self.on_startup = [ register_uservice() ]
```

Comment:
See above comment.

A note about this line of code: we don't need to round brackets here as we want to pass the function in as a function object, rather than call the function directly.